libs/source/Source.ml

```diff
@@ -0,0 +1,48 @@
+(* Cooper Pierce
+ *
+ * Copyright (c) Semgrep Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+
+(* See Source.mli for top-level documentation of this module. *)
+
+type t = File of Fpath.t [@@deriving show, eq, ord]
+
+(* Module for the purpose of implementing sexp conversions for t. We cannot
+ * derive sexp since Fpath does not have an sexp implementation, so we convert
+ * to a string and then derive sexp on the updated type.
+ *)
```

Comment:
What are S-expressions used for? I'd rather not depend on yet another format.

Reply:
I believe at some point this was getting transitively relied on somehow for the libdef serialization (which uses s-expressions). Not 100% sure through what, since I just tried to find it and it seems like it is no longer needed. Given that, I'll remove it.

Reply:
Done in 5fee815255b446b1e3360b63d94021332adf6f60